app/tests/archives_tests/test_views.py

```diff
@@ -1343,3 +1345,21 @@ def make_request():
     with django_assert_num_queries(expected_queries):
         resp = make_request()
         assert len(resp.json()["data"]) == 4
+
+
+@pytest.mark.django_db
+def test_job_list_row_template_renders(
+    client, settings, django_capture_on_commit_callbacks
+):
+    job = AlgorithmJobFactory()
+
+    assert job.algorithm_image
+
+    job.status = Job.SUCCESS
+
+    response = render_to_string(
```

Comment:
These two tests only test that they're rendering here, not that they're rendering correctly in our application. You cannot be sure that these templates will be used by the view that was broken. You cannot be sure that the template was correctly parametrised. To test a view you need to use the client to make a call.

Reply:
The case here is that the call to render the table rows in the view is asyncronous (made as an AJAX call by JQuery/datatables). So, when testing the view with the client call, it will not fail, since the actual problematic rendering happens in the AJAX call. Therefore, I thought of testing the rendering of the row template with an job object in the context and check for the presence of the button (i.e. rendered successfully) in the output string. 

How do you propose testing for dynamic Javascript using Django view client calls?

Reply:
Make the same call that the AJAX request makes.